test/tstQueryTreeRay.cpp

```diff
@@ -131,4 +154,54 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(test_ray_box_intersection, DeviceType,
       make_reference_solution<int>({0, 1, 2, 3, 4, 5, 6, 7, 8, 9}, {0, 10}));
 }
 
+template <typename DeviceType>
+struct InsertIntersections
+{
+  mutable int _i;
+  Kokkos::View<int *, DeviceType> _ordered_intersections;
+
+  template <typename Predicate>
+  KOKKOS_FUNCTION void operator()(Predicate const &,
+                                  int const primitive_index) const
+  {
+    _ordered_intersections(_i++) = primitive_index;
```

Comment:
The incrementing does not need atomicity?

Reply:
There is only one predicate and every predicate is processed by a single thread so it's not really necessary to use atomics here.

Reply:
this is a bug begging to happen

Reply:
I mean it's still a variable local to the predicate even if there were more but sure I'll make it atomic, see [03e716f](https://github.com/arborx/ArborX/pull/683/commits/03e716fd2a379b930005071710e677658ae50059)